File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

class Hamming
  def self.compute(a, b)
    raise ArgumentError.new if a.length != b.length
    a, b = a.split(''), b.split('')
    c = a.zip(b)
    sum = 0
    c.each{|i| sum += 1 if i[0] != i[1]}
    return sum
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(a, b) {
  if (a.length != b.length) {
    throw new Error('DNA strands must be of equal length.');
  } else {
    var sum = 0;
    for (var i = 0; i < a.length; i++) {
      if (a[i] != b[i]) {
        sum++;
      }
    }
    return sum;
  }
};

module.exports = Hamming;